lib/utils.js

```diff
@@ -27,7 +27,7 @@ export function isMajorRange(range) {
 }
 
 export function isMaintenanceRange(range) {
-  return /^\d+\.(?:\d+|x)(?:\.x)?$/i.test(range);
+  return /^\d+(?:\.(\d+|x))?(?:\.x)$/i.test(range);
```

Comment:
```suggestion
  return /^\d+(?:\.(?:\d+|x))?\.x$/i.test(range);
```

lets keep the groups as non-capturing. also, the last segment no longer needs to be a group since it is no longer optional

Reply:
Why should we use non-capturing groups? Is it just because of better performance or is there another reason?

Reply:
just no need to do any capturing. mostly thinking about consistency so that we dont wonder in the future why one group is non-capturing and one group is capturing within a regex used for just testing if there is a match or not. i tend to get rusty with regex, so removing potential questions for future maintenance is my main goal. again, open to suggestions if there are reasons to do anything beyond what i'm thinking about

Reply:
> so removing potential questions for future maintenance is my main goal

Got it! Thank you for the explanation. I have nothing to add. :+1: